spark/v3.5/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestCreateChangelogViewProcedure.java

```diff
@@ -242,6 +249,34 @@ public void testUpdateWithIdentifierField() {
         sql("select * from %s order by _change_ordinal, id, data", viewName));
   }
 
+  @TestTemplate
+  public void testUpdateWithInComparableType() {
+    createTableWithIncomparableType();
+
+    sql("INSERT INTO %s VALUES (2, Map('b','b'), 12)", tableName);
+    Table table = validationCatalog.loadTable(tableIdent);
+    Snapshot snap1 = table.currentSnapshot();
+
+    sql("INSERT OVERWRITE %s VALUES (3, Map('c','c'), 13), (2, Map('d','d'), 12)", tableName);
```

Comment:
Doesn't  generating change logs rows depend on the ordering of the rows?
If the ordering is removed on specific column, it can potentially  return incorrect results?
For example if the first column type was Map, we would sort on the remaining columns. And we might not be able to detect chnages reliably?

Reply:
Let me trying to think out loud
- Given the identifier column is missing, we are basically sort all columns to generate changelog, here we are skip sorting on the columns which cannot be ordered. I dont feel like we will lose on detect the change as we did not filter any row out in the results
- If identifier columns are added to the table or provided in procedure explicitly, [the repartition spec will only include such instead of all columns](https://github.com/apache/iceberg/blob/main/spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/procedures/CreateChangelogViewProcedure.java#L164-L176) and later used for sorting

Reply:
I can see code where [we compare every column of two rows](https://github.com/apache/iceberg/blob/main/spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/ChangelogIterator.java#L101) in a iterator  to see if they are equal and base that to generate the change log.

For example consider a table with schema <id: Int, map: Map<String, String>> having records

` 1, Map<a, a>`, `1, Map<b, b>`,  `1, Map<a, a>`

An ordering based on `id`
can result in in 
```
 1, Map<a, a>
 1, Map<a, a>
 1, Map<b, b>
```
In this case first two rows are same
or
```
record1 => 1, Map<a, a>
record2 => 1, Map<b, b>
record3 => 1, Map<a, a>
```
In this case first two rows are different. 
Hence I was curious if the result would differ when we skip the ordering on a column.

Reply:
Exactly what I thought. We can't skip the ordering for a specific column because any non key column are under DML consideration. It can impact the net changes outcome. @flyrain what you say?

Reply:
That's valid concern. Skipping certain columns breaks the [assumption](https://github.com/apache/iceberg/blob/79620e198009fa243c278c66fd442d107b46206a/spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/RemoveCarryoverIterator.java#L28-L28) of `ChangelogIterator`. I'd suggest to start from the class `ChangelogIterator` to see how it works with dataset having certain columns not sorted.